coco/request_forwarder.py

```diff
@@ -265,14 +270,22 @@ async def _request(self, session, method, host, endpoint, request, params):
                     return host, (await response.text(), response.status)
         except AsyncioTimeoutError:
             self.call_counter.labels(
-                endpoint=endpoint, host=hostname, port=port, status="0"
+                endpoint=endpoint, host=hostname, port=port, status=status
             ).inc()
             return host, ("Timeout", 0)
         except Exception as e:
             self.call_counter.labels(
-                endpoint=endpoint, host=hostname, port=port, status="0"
+                endpoint=endpoint, host=hostname, port=port, status=status
             ).inc()
             return host, (str(e), 0)
+        finally:
+            response_time = time.time() - start_time
+            self.response_time.labels(
+                endpoint=endpoint, host=hostname, port=port
+            ).observe(response_time)
+            self.call_counter.labels(
```

Comment:
If there is an exception, `call_counter` gets incremented twice?

Reply:
Oh nvm, in the `except` blocks there is a `return`. So if there is an `exception`, we will get one `call_counter`, and no `response_time` (which makes sense. Okay.

Reply:
No, I think regardless of the `return`, `finally` will get executed, so we should  remove the `call_counter` from there.

Reply:
```
 def func():
     try:
         2 + 'hello'
     except:
         print('1')
         return
     finally:
         print('2')
 func()    
 ```
 ```
 anjakefala@chime:~$ python3 test.py 
1
2
```

You are right

Reply:
Or move it to within the `try` block, after the statement that would normally trigger the exception.